### PR TITLE
1322 reproductive health facility spot check rdd change

### PR DESCRIPTION
**Pull request by billli0**
### Issue #:
[1322: UNFPA: Reproductive health facility spot check RDD changes](https://github.com/beyondessential/tupaia-backlog/issues/1322)
### Changes:

- In custom raw data download, shorten sheet name if more than 31 chars 
- Convert service_type from dhis to tupaia in this survey's data elements
- Convert survey from raw data download to custom download

### CODE REVIEW

File: packages/database/src/migrations/20201126234247-ConvertDhisSurveysIntoTupaia-modifies-data.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+};
+
+const surveyCodes = ['RHC', 'RHFSC', 'RHFSCTS'];
+
+const switchServiceType = async (db, code, serviceType) => {
+  await updateValues(db, 'data_source', { service_type: serviceType }, { code });
+};
+
+const getDataSources = async (db, codes) => {
+  const surveyCodeQuery = codes.map(e => `'${e}'`).join(',');
```
Comment (kael89) [inline packages/database/src/migrations/20201126234247-ConvertDhisSurveysIntoTupaia-modifies-data.js:27]: No need to change your code here, but there's also `arrayToDbString` in the migration utilities which does the same thing - we use that a lot!

File: packages/database/src/migrations/20201126234247-ConvertDhisSurveysIntoTupaia-modifies-data.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  return db.runSql(`
+  select ds.id, ds.code from data_source ds 
+    join data_element_data_group dedg on ds.id = dedg.data_element_id 
+    join data_source ds2 on ds2.id = dedg.data_group_id 
+  where ds2.code in (${surveyCodeQuery}) 
+  group by (ds.id, ds.code)
```
Comment (kael89) [inline packages/database/src/migrations/20201126234247-ConvertDhisSurveysIntoTupaia-modifies-data.js:33]: The grouping here is clever! 💯

File: packages/database/src/migrations/20201126234247-ConvertDhisSurveysIntoTupaia-modifies-data.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  `);
+};
+
+exports.up = async function (db) {
+  const { rows: dataSources } = await getDataSources(db, surveyCodes);
+  const serviceType = 'tupaia';
+
+  for (const dataSource of dataSources) {
+    await switchServiceType(db, dataSource.code, serviceType);
+  }
```
Comment (kael89) [inline packages/database/src/migrations/20201126234247-ConvertDhisSurveysIntoTupaia-modifies-data.js:43]: You could also do a mass update here, using a custom `UPDATE` query:

```sql
UPDATE data_source SET service_type = 'tupaia' WHERE code IN (...your codes here...)
```

But don't worry about it, unless if executing the above query for 1000+ records takes significant time

File: packages/database/src/migrations/20201201003822-ConvertSurveyRawDataDownloadToCustomDataDownload-modifies-data.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+            entityAggregation: {
+              dataSourceEntityType: 'facility',
+            },
+          },
+        },
+        transformations: [
+          {
+            type: 'transposeMatrix',
+          },
+        ],
```
Comment (kael89) [inline packages/database/src/migrations/20201201003822-ConvertSurveyRawDataDownloadToCustomDataDownload-modifies-data.js:58]: This looks correct but I would like to ask @nick-beyondessential to place his `surveyDataExport` expert eyes on this

**Review by kael89 — APPROVED** [review]
Great work @billli0 ! Just a question and a couple of comments but this looks correct overall. Approving, but it would be good to get @nick-beyondessential 's approval of a piece of code that he is more familiar with

File: packages/database/src/migrations/20201201003822-ConvertSurveyRawDataDownloadToCustomDataDownload-modifies-data.js (hunk 4 above)
Comment (nick-beyondessential) [reply packages/database/src/migrations/20201201003822-ConvertSurveyRawDataDownloadToCustomDataDownload-modifies-data.js:58]: Perfect!

File: packages/web-config-server/src/export/ExportSurveyDataHandler.js [hunk 5]
```diff
@@ -73,8 +74,10 @@ export class ExportSurveyDataHandler extends RouteHandler {
         origin: 'A2',
       });
 
-      sheetNames.push(surveyName);
-      sheets[surveyName] = sheet;
+      const sheetName = truncateString(surveyName, 31);
```
[the remark this answers is not in the document]
Comment (billli0) [reply packages/web-config-server/src/export/ExportSurveyDataHandler.js:77]: Yes it is! It happens in Meditrak server as well. It is only one line of code so not sure if we need to capsulize them as util.

File: packages/database/src/migrations/20201126234247-ConvertDhisSurveysIntoTupaia-modifies-data.js (hunk 3 above)
Comment (billli0) [reply packages/database/src/migrations/20201126234247-ConvertDhisSurveysIntoTupaia-modifies-data.js:43]: Clever!

File: packages/web-config-server/src/export/ExportSurveyDataHandler.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (kael89) [reply packages/web-config-server/src/export/ExportSurveyDataHandler.js:77]: I believe it's fine not to do so at this stage. We could create and use a generic `SpreadsheetExporter` in out `utils` package, but that's a job for a different day

**Pull request closed without merging**